packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -568,7 +568,14 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 				};
 				const summarizeResult = this.generator.summarize(summaryOptions, resultsBuilder);
 				// ensure we wait till the end of the process
-				return summarizeResult.receivedSummaryAckOrNack;
+				const result = await summarizeResult.receivedSummaryAckOrNack;
```

Comment:
`trySummarizeOnce` is not part of the heuristics based summary right. Do we need to log failure here? We only care about the heuristcs based summary right.

Reply:
I think the idea is that anyone who runs a summary should send a `"SummarizeFailed"` event on failure of all retries

Reply:
Actually, I just realized this is called for last summary flow too, so this is good.